src/aten/sycl/Col2ImKernel.cpp

```diff
@@ -0,0 +1,259 @@
+      for (int64_t w_col = w_col_start; w_col < w_col_end; w_col += 1) {
+        int64_t h_k = (h_im - h_col * stride_h);
+        int64_t w_k = (w_im - w_col * stride_w);
+        if (h_k % dilation_h == 0 && w_k % dilation_w == 0) {
+          h_k /= dilation_h;
+          w_k /= dilation_w;
+          int64_t data_col_index =
+              (((c_im * kernel_h + h_k) * kernel_w + w_k) * output_height +
+               h_col) *
+                  output_width +
+              w_col;
+          val += in_ptr[data_col_index];
+        }
+      }
+    }
+    out_ptr[id] = static_cast<T>(val);
+  }
+  Col2imKernelFunctor(
+      const T* in_data_,
+      const int64_t channels_,
+      const int64_t height_,
+      const int64_t width_,
+      const int64_t output_height_,
+      const int64_t output_width_,
+      const int64_t kernel_h_,
+      const int64_t kernel_w_,
+      const int64_t pad_h_,
+      const int64_t pad_w_,
+      const int64_t stride_h_,
+      const int64_t stride_w_,
+      const int64_t dilation_h_,
+      const int64_t dilation_w_,
+      T* out_data_)
+      : in_data(in_data_),
+        channels(channels_),
+        height(height_),
+        width(width_),
+        output_height(output_height_),
+        output_width(output_width_),
+        kernel_h(kernel_h_),
+        kernel_w(kernel_w_),
+        pad_h(pad_h_),
+        pad_w(pad_w_),
+        stride_h(stride_h_),
+        stride_w(stride_w_),
+        dilation_h(dilation_h_),
+        dilation_w(dilation_w_),
+        out_data(out_data_) {}
+
+ private:
+  const T* in_data;
+  const int64_t channels;
+  const int64_t height;
+  const int64_t width;
+  const int64_t output_height;
+  const int64_t output_width;
+  const int64_t kernel_h;
+  const int64_t kernel_w;
+  const int64_t pad_h;
+  const int64_t pad_w;
+  const int64_t stride_h;
+  const int64_t stride_w;
+  const int64_t dilation_h;
+  const int64_t dilation_w;
+  T* out_data;
+};
+
+template <typename T>
+static void col2im_kernel(
+    const T* data_col,
+    const int64_t channels,
+    const int64_t height,
+    const int64_t width,
+    const int64_t output_height,
+    const int64_t output_width,
+    const int64_t kernel_h,
+    const int64_t kernel_w,
+    const int64_t pad_h,
+    const int64_t pad_w,
+    const int64_t stride_h,
+    const int64_t stride_w,
+    const int64_t dilation_h,
+    const int64_t dilation_w,
+    T* data_im) {
+  auto& sycl_queue = at::xpu::getCurrentSYCLQueue();
+  auto total_threads = channels * width * height;
+
+  auto in_data = data_col;
+  auto out_data = data_im;
+  auto kfn = Col2imKernelFunctor<T>(
+      in_data,
+      channels,
+      height,
+      width,
+      output_height,
+      output_width,
+      kernel_h,
+      kernel_w,
+      pad_h,
+      pad_w,
+      stride_h,
+      stride_w,
+      dilation_h,
+      dilation_w,
+      out_data);
+  sycl_kernel_submit(total_threads, sycl_queue, kfn);
+}
+
+void col2im_out_template(
```

Comment:
Do we consider to refine the cuda/cpu templates, instead of copy out?

Reply:
Sure. We can use **#include <ATen/native/im2col_shape_check.h>** instead of current.